src/unitxt/formats.py

```diff
@@ -200,43 +206,72 @@ def process(
         return instance
 
 
-class HFSystemFormat(BaseFormat):
-    r"""Formats the complete input for the model using the HuggingFace chat template of a given model.
+class TextContent(TypedDict):
+    type: Literal["text"]
+    text: str
 
-    HFSystemFormat expects the input instance to contain:
-    1. A field named "system_prompt" whose value is a string (potentially empty) that delivers a task-independent opening text.
-    2. A field named "source" whose value is a string verbalizing the original values in the instance (as read
-    from the source dataset), in the context of the underlying task.
-    3. A field named "instruction" that contains a (non-None) string.
-    4. A field named with the value in arg 'demos_field', containing a list of dicts, each dict with fields "source"
-    and "target", representing a single demo.
-    5. A field named "target_prefix" that contains a string to prefix the target in each demo, and to end the whole generated prompt.
 
-    SystemFormat formats the above fields into a single string to be inputted to the model. This string overwrites
-    field "source" of the instance.
+class ImageUrlContent(TypedDict):
+    type: Literal["image_url"]
+    image_url: Dict[Literal["url"], str]
 
-    Example:
-        HFSystemFormat(model_name="HuggingFaceH4/zephyr-7b-beta")
 
-        Uses the template defined the in tokenizer_config.json of the model:
+class ImageFileContent(TypedDict):
+    type: Literal["image_file"]
+    image_file: Dict[Literal["file_id"], str]
 
-        "chat_template": "{% for message in messages %}\n{% if message['role'] == 'user' %}\n{{ '<|user|>\n' + message['content'] + eos_token }}\n{% elif message['role'] == 'system' %}\n{{ '<|system|>\n' + message['content'] + eos_token }}\n{% elif message['role'] == 'assistant' %}\n{{ '<|assistant|>\n'  + message['content'] + eos_token }}\n{% endif %}\n{% if loop.last and add_generation_prompt %}\n{{ '<|assistant|>' }}\n{% endif %}\n{% endfor %}",
 
-        See more details in https://huggingface.co/docs/transformers/main/en/chat_templating
+Content = Union[TextContent, ImageUrlContent, ImageFileContent]
 
-    """
 
-    model_name: str
-    _requirements_list = ["transformers"]
+class Message(TypedDict):
+    role: Literal["system", "user", "assistant"]
+    content: Union[str, List[Content]]
 
-    def prepare(self):
-        from transformers import AutoTokenizer
 
-        self.tokenizer = AutoTokenizer.from_pretrained(self.model_name)
+class OpenAIFormat(BaseFormat):
```

Comment:
We need documentation. About the output, the use case and so on. I think I understand what you meant to do, but I'm not sure it generalize well... lets make sure by first you writing what you meant to do

Reply:
done

Reply:
write what this format is. Give an example. If there are multiple formats list them